### PR TITLE
Independent e2e tests

### PR DESCRIPTION
**Pull request by sgnn7**
### What does this PR do?
- Makes E2E Linux tests cover all general configuration permutations

### What ticket does this PR close?
Connected to #101 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [x] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: examples/puppetmaster/test.sh [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+  echo "-----"
+  run_in_conjur conjur policy load root /src/policy.yml
+  run_in_conjur conjur variable values add inventory/db-password $EXPECTED_PASSWORD  # load the secret's value
+}
+
+revoke_cert_for() {
+  local cert_fqdn="$1"
+  echo "Ensuring clean cert state for $cert_fqdn..."
+
+  # Puppet v5 and v6 CLIs aren't 1:1 compatible so we have to chose the format based
```
Comment (diverdane) [inline examples/puppetmaster/test.sh:150]: Typo: s/chose/choose/

File: examples/puppetmaster/test.sh [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+
+  local config_file="$TMPDIR/conjur.conf"
+  local identity_file="$TMPDIR/conjur.identity"
+
+  echo "
+    appliance_url: https://conjur-https:$CONJUR_SERVER_PORT/
+    version: 5
+    account: cucumber
+    cert_file: /etc/ca.crt
+  " > $config_file
```
Comment (diverdane) [inline examples/puppetmaster/test.sh:182]: This is probably a don't care, but I think this leaves an extra line with 2 spaces at the bottom of the config file. Why not use heredoc format? Same question for other places where configs are being generated.

File: examples/puppetmaster/policy.yml [hunk 3]
```diff
@@ -13,11 +14,31 @@
...
+      annotations:
+        puppet: true
+
+    - !host &h_manifest_apikey
+      id: /manifest-apikey-node
+      annotations:
+        puppet: true
+
+    # Vagrant test node
+    - !host &h_node01
```
[the remark this answers is not in the document]
Comment (sgnn7) [reply examples/puppetmaster/policy.yml:33]: Yes absolutely - I just didn't want to start changing those tests too!

File: examples/puppetmaster/test.sh (hunk 2 above)
Comment (sgnn7) [reply examples/puppetmaster/test.sh:182]: This is old code but I think the format here makes the logic easier to read and it's still a valid conf file even if it's a bit of an odd spacing within the file

**Pull request merged**